### PR TITLE
feat(storage-resize-images):wildcard in include/exclude paths

### LINKED ISSUE #560: Use wildcards in paths for resizing images
### Extension name

storage-resize-images

### What feature would you like to see?

The ability to use glob-like wildcards in paths for specifying the location of images we want to resize.

E.g. "/user/*/images" will match the image subdirectory of all users (E.g. /user/a2/images, /user/b3/images etc)

Also support ** for directory matching at any depth.

### How would you use it?

As with documents stored in Firestore, where the pathing structure denotes the logical informational hierarchy, the storage bucket paths normally follow the same structure (for storing large objects associated with the Firestore documents). In the example above, storing a user's personal account images.

As it currently stands, I can't make use of this handy extension.

### PR DESCRIPTION
**Pull request by russellwheatley**
fixes #560

Allow wildcard notation for directory names.

Supplied path for images I want to be resized: `/users/*/pictures`

## Include path tests

Unsuccessful upload to `/users/portraits/avatar`:
Screenshot 2021-01-19 at 16 02 46

Uploaded successfully to `/users/profile`:
Screenshot 2021-01-19 at 16 00 44
/pictures`:

Supplied paths for images I don't want to be resized: `/animals/*/cats,/animals/*/dogs/*/sleepy`

## Exclude path tests

Unsuccessful upload to `/animals/fluffy/dogs/large/sleepy`:
Screenshot 2021-01-19 at 16 05 16

Unsuccessful upload to `/animals/fluffy/cats/large/sleepy`:
Screenshot 2021-01-19 at 16 12 14

Uploaded successfully to `/animals/fluffy/dogs/large/fast`:
Screenshot 2021-01-19 at 16 09 18

### CODE REVIEW

**Comment by russellwheatley** [comment]
Hey @kevinthecheung, do you mind going over the documentation aspect, please? 😄

**Review by dackers86 — APPROVED** [review]

**Comment by dackers86** [comment]
## Testing

### Ignore paths

image

image

image


### Include paths

image

image

**Comment by dackers86** [comment]
@googlebot I consent

**Review by Salakar — APPROVED** [review]

**Comment by dackers86** [comment]
Hi @kevinthecheung.

Would it be possible to add this to your review list. Thanks!

File: storage-resize-images/extension.yaml [hunk 1]
```diff
@@ -172,9 +172,12 @@ params:
...
     description: >
-      Restrict storage-resize-images to only resize images in specific locations in your Storage bucket by 
-      supplying a comma-separated list of absolute paths. For example, to only resize the images 
-      stored in `/users/pictures` directory, specify the path `/users/pictures`. 
+      A comma-separated list of absolute paths. Restrict image resizes to 
+      the supplied paths. The path only has to match initially for it to be resized.
+      For example, if you supplied this path `/users/pictures`, it will resize images 
+      in  `/users/pictures/image.png` & `/users/pictures/any/sub/directory/image.png`.
+      You may also used wildcard notation for directories in the path e.g `user/*/pictures` 
+      would match `users/profile/pictures/image.png` & `users/profile/pictures/any/sub/directory/image.png`
```
Comment (jhuleatt) [inline storage-resize-images/extension.yaml:180]:
```suggestion
      Restrict storage-resize-images to only resize images in specific locations in your Storage bucket by 
      supplying a comma-separated list of absolute paths. For example, to only resize the images 
      stored in the `/users/pictures` directory, specify the path `/users/pictures`.
      
      You may also use wildcard notation for directories in the path. For example, `user/*/pictures` 
      would match `users/profile/pictures/image.png` as well as  `users/profile/pictures/any/sub/directory/image.png`. 
```

File: storage-resize-images/extension.yaml [hunk 2]
```diff
@@ -172,9 +172,12 @@ params:
...
-      supplying a comma-separated list of absolute paths. For example, to only resize the images 
-      stored in `/users/pictures` directory, specify the path `/users/pictures`. 
+      A comma-separated list of absolute paths. Restrict image resizes to 
+      the supplied paths. The path only has to match initially for it to be resized.
+      For example, if you supplied this path `/users/pictures`, it will resize images 
+      in  `/users/pictures/image.png` & `/users/pictures/any/sub/directory/image.png`.
+      You may also used wildcard notation for directories in the path e.g `user/*/pictures` 
+      would match `users/profile/pictures/image.png` & `users/profile/pictures/any/sub/directory/image.png` 
       If you prefer to resize every image uploaded to the storage bucket, 
       leave this field empty.
```
Comment (jhuleatt) [inline storage-resize-images/extension.yaml:182]:
```suggestion
      Restrict storage-resize-images to only resize images in specific locations in your Storage bucket by 
      supplying a comma-separated list of absolute paths. For example, to only resize the images 
      stored in the `/users/pictures` directory, specify the path `/users/pictures`.
      
      You may also use wildcard notation for directories in the path. For example, `users/*/pictures` 
      would match `users/profile/pictures/image.png` as well as  `users/profile/pictures/any/sub/directory/image.png`. 
      
      If you prefer to resize every image uploaded to the storage bucket, 
      leave this field empty.
```

**Review by jhuleatt — APPROVED** [review]

**Comment by jhuleatt** [comment]
Chatted with Kevin and applied some docs changes

File: storage-resize-images/extension.yaml [hunk 3]
```diff
@@ -174,7 +174,11 @@ params:
     description: >
       Restrict storage-resize-images to only resize images in specific locations in your Storage bucket by 
       supplying a comma-separated list of absolute paths. For example, to only resize the images 
-      stored in `/users/pictures` directory, specify the path `/users/pictures`. 
+      stored in the `/users/pictures` and `/restaurants/menuItems` directories, specify the paths `/users/pictures,/restaurants/menuItems`.
+      
+      You may also use wildcard notation for directories in the path. For example, `/users/*/pictures` 
+      would match `/users/profile/pictures/image.png` as well as  `/users/profile/pictures/any/sub/directory/image.png`. 
+      
       If you prefer to resize every image uploaded to the storage bucket,
```
Comment (jhuleatt) [inline storage-resize-images/extension.yaml:182]:
```suggestion
      If you prefer to resize every image uploaded to your Storage bucket, 
```

**Review by kevinthecheung — APPROVED** [review]

**Pull request merged**